pkg/azure/aks/pricing_map.go

```diff
@@ -0,0 +1,183 @@
+package aks
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	retailPriceSdk "gomodules.xyz/azure-retail-prices-sdk-for-go/sdk"
+)
+
+const (
+	AZ_API_VERSION string = "2023-01-01-preview" // using latest API Version https://learn.microsoft.com/en-us/rest/api/cost-management/retail-prices/azure-retail-prices
+)
+
+type MachineOperatingSystem int
+
+const (
+	Linux MachineOperatingSystem = iota
+	Windows
+)
+
+func (o MachineOperatingSystem) String() string {
+	return [2]string{"Linux", "Windows"}[o-1]
+}
+
+type MachinePriority int
+
+const (
+	OnDemand MachinePriority = iota
+	Spot
+)
+
+func (v MachinePriority) String() string {
+	return [2]string{"OnDemand", "Spot"}[v-1]
```

Comment:
IMO I would instantiate the slice of strings as a variable so there isn't an allocation made every call to the String method.

Reply:
Sounds good.  Updated.